packages/compiler-core/src/transforms/transformElement.ts

```diff
@@ -594,8 +594,6 @@ export function buildProps(
         hasDynamicKeys = true
         if (exp) {
           if (isVBind) {
-            // #10696 in case a v-bind object contains ref
-            pushRefVForMarker()
             // have to merge early for compat build check
             pushMergeArg()
```

Comment:
Given the new call to `pushMergeArg()` that's been added just before `mergeArgs.push(exp)`, I'm wondering whether this earlier call can now be moved inside the `if (__COMPAT__) {` check?

For non-compat builds I think that'd be equivalent to what we have currently, with just a single call to `pushMergeArg()`, so we'd only have the double push in the compat build.

Reply:
>  I'm wondering whether this earlier call can now be moved inside the if (__COMPAT__) { check?

I've updated it via https://github.com/vuejs/core/pull/12993/commits/51fb524a61b726e3f21984eb060da952d8876313